blog-pipeline/tests/test_dry_run.py

```
@@ -0,0 +1,710 @@
+_SKIP_MSG = "Test data not available"
+
+
+class TestABulletNesting(unittest.TestCase):
+    """Test A: Bullet nesting — 2-space indent doubling and nested <ul> generation."""
+
+    @unittest.skipIf(not _DATA_AVAILABLE, _SKIP_MSG)
+    def test_fix_bullet_indent_doubles_2space(self):
+        text = ARTICLE_FINAL.read_text(errors="replace")
+        # Find lines with 2-space indent bullets
+        two_space_lines = [
+            l for l in text.split("\n")
```

Ruff (0.15.7)

[error] Ambiguous variable name: `l`

(E741)